Compilation warning for test suite

Compiling the test suite gives me this warning:
```
[26/319] Building CXX object CMakeFiles/s2builderutil_lax_polygon_layer_test.dir/src/s2/s2builderutil_lax_polygon_layer_test.cc.o
../src/s2/s2builderutil_lax_polygon_layer_test.cc: In function ‘std::string {anonymous}::ToString(DegenerateBoundaries)’:
../src/s2/s2builderutil_lax_polygon_layer_test.cc:56:1: warning: control reaches end of non-void function [-Wreturn-type]
   56 | }
      | ^
[319/319] Linking CXX executable s2lax_polygon_shape_test
```

This is because there is a switch here to convert enum values to strings, but the switch has no default option (for cases in which the enum has a value the switch doesn't pick up on).

--- a/src/s2/s2builderutil_lax_polygon_layer_test.cc
+++ b/src/s2/s2builderutil_lax_polygon_layer_test.cc
@@ -53,6 +53,10 @@ string ToString(DegenerateBoundaries degenerate_boundaries) {
     case DegenerateBoundaries::DISCARD_SHELLS: return "DISCARD_SHELLS";
     case DegenerateBoundaries::KEEP: return "KEEP";
   }
+  // Cases are exhaustive, but some compilers don't know that and emit a
+  // warning.
+  S2_LOG(FATAL) << "Unknown DegenerateBoundaries value: "
+                << static_cast<int>(degenerate_boundaries);
 }
 
 void TestLaxPolygon(string_view input_str, string_view expected_str,

